### DISCUSSION TITLE
Option to run cells with only with `mo.md()` on startup?

### DISCUSSION THREAD

**Discussion by roberthobblebottom** [marimo-team/marimo#603] [Ideas]
In the settings I have set `autorun on startup` in the settings as unchecked, which is good as expensive cells that I forgotten to disable and it ended up real world money (OpenAI api) but I wonder if there could be another option to run cells with only with `mo.md()` on startup?

**Comment by mscolnick** [comment]
This is a bit tricky since there are some edge-cases. 

1) your markdown could be an f'string with expensive substitutions. e.g. `mo.md(f"Total count: {query_db()})`
2) your imports could be expensive as well. the cell with `import marimo as mo`, could also be expensive. 

We could probably solve 1 (only run when the dependency is just `marimo`).
For 2, would that defeat the purpose of `autorun on startup`? We could give another option for `autorun markdown on startup` which includes imports and "pure" `mo.md`

**Comment by alexscordellis** [comment]
I'd be very keen on this - and to avoid edge-case 2 I would be happy for it to be gated behind the cell that contains `import marimo as mo` having nothing but imports in it (or some similar condition).

My use case is that I want the Outline view to work on a notebook that has not fully executed. My most common use case is to have quite large notebooks with various sections, each of which has been used to perform some ad-hoc or prototyped data analysis, much of which will be expensive/slow. So I want to be able to open an old notebook and use the Outline view to jump to the right section or remind myself what's in the notebook, without having to recompute everything.

FWIW marimo has come on so far in the last few months that not only have I entirely replaced Jupyter with it, but this niggle is just about the only thing I miss from Jupyter any more, great work.

**Comment by stephenlf** [comment]
I have gotten into the habit of putting expensive or non-idempotent code behind a run button. It's a couple of lines of boilerplate, but it ends up being a nicer experience overall. Couple that with some accordions to spit out results, and you get an outline and run log in one downloadable file. 

Set up a run button
```python
# Set up run button
run_button = mo.ui.run_button(label="Do expensive thing")
```

Gate the expensive calculations behind `run_button.value`
```python
# optionally, annotate w/ mo.cache
def do_expensive():
    if run_button.value:  # Key line
        # ...do the thing
        pass

run_result = do_expensive()
```

Embed the run button and results in the markdown outline
```md
# This is a Markdown Outline
Lorem ipsum etc.

Click here to run the calculation:
{run_button}

And here are the results!
{mo.accordion({"Run Result": run_result})}

Isn't that nice?
```
